visualizer/draw/area-chart.js

```diff
@@ -381,29 +448,62 @@ class AreaChart extends HtmlContent {
       .call(xAxis)
 
     this.pixelsPerSlice = usableWidth / this.slicesCount
+    this.pixelsPerStack = usableHeight / this.stacksCount
 
     this.d3SliceHighlight.style('width', Math.max(this.pixelsPerSlice, 1) + 'px')
     this.d3SliceHighlight.style('height', usableHeight + 'px')
     this.d3SliceHighlight.style('top', margins.top + 'px')
   }
 
-  drawFiltering () {
-    this.d3AreaPaths.classed('filtered', d => d.key.split('_')[1] === 'absent' || (this.layoutNode && this.layoutNode.id !== extractLayoutNodeId(d.key)))
-  }
-
   draw () {
     super.draw()
-
     if (this.layoutNodeToApply) {
       this.layoutNode = this.layoutNodeToApply
       this.layoutNodeToApply = null
-      this.drawFiltering()
     }
     if (this.widthToApply && this.widthToApply !== this.width) {
       this.drawPathsToFit(this.widthToApply)
     }
+    this.drawToCanvas()
     if (!this.isInHoverBox) this.widthToApply = null
   }
+
+  clearCanvas () {
+    this.canvasContext.clearRect(0, 0, this.d3CanvasPlotArea.attr('width'), this.d3CanvasPlotArea.attr('height'))
+  }
+
+  drawToCanvas () {
+    const areas = this.dataContainer.selectAll('custom.area')
+    if (!this.highlightedLayoutNode) {
+      this.clearCanvas()
+      areas.each(d => this.drawNodeArea(d))
+    } else {
+      if (this.isInHoverBox) {
+        this.clearCanvas()
+      }
+      if (this.currentHighlightedArea) {
+        this.currentHighlightedArea.each(d => this.drawNodeArea(d))
+      }
+      this.currentHighlightedArea = areas.select((d, i, nodes) => this.highlightedLayoutNode.id === extractLayoutNodeId(d.key) ? nodes[i] : null)
+      this.currentHighlightedArea.each(d => this.drawNodeArea(d))
+    }
+  }
+
+  drawNodeArea (d) {
+    const cssId = `type-${d.key.split('_')[0]}`
+    const isFiltered = d.key.split('_')[1] === 'absent' || (this.layoutNode && this.layoutNode.id !== extractLayoutNodeId(d.key))
+    const isEven = !(d.index % 2)
+    const isHighlighted = this.highlightedLayoutNode && extractLayoutNodeId(d.key) === this.highlightedLayoutNode.id
+    const { fillColour, opacity } = this.getCanvasAreaStyles(cssId, isEven, isFiltered, isHighlighted)
+    const d3Col = d3.color(fillColour)
+    d3Col.opacity = opacity
+    this.canvasContext.beginPath()
+    this.areaMaker.context(this.canvasContext)(d)
+    this.canvasContext.fillStyle = this.getCSSVarValue('--main-bg-color')
```

Comment:
Probably worth adding a comment here so it's clear why we're doing this twice, like:

```suggestion
    // Reset area's colour, so repeated hover in/out doesn't overlay colours increasing their intensity
    this.canvasContext.fillStyle = this.getCSSVarValue('--main-bg-color')
```

Reply:
There's also a rare feature-bug here (opposite of a bug-feature). 

Caching the CSS variables like this means the value of `--main-bg-color` used stays light if the user switches to the light theme, so it's painting green, purple etc on dark grey not light grey. 

But, this arguable slight regression is currently an accidental enhancement because those colours were a bit too light for the light theme, so the chart's slightly more usable in the light theme this way... So for now let's leave it in but I'll add a ticket for a separate task to use slightly darker colours in the the light theme and clear the CSS variable cache when the theme changes. Maybe also, move it to the general scope as I think there are other parts of the code that might benefit from using it.